starter/cloudstorage/src/main/java/com/udacity/jwdnd/course1/cloudstorage/controller/FileController.java

```diff
@@ -0,0 +1,92 @@
+package com.udacity.jwdnd.course1.cloudstorage.controller;
+
+import com.udacity.jwdnd.course1.cloudstorage.data.File;
+import com.udacity.jwdnd.course1.cloudstorage.data.FileForm;
+import com.udacity.jwdnd.course1.cloudstorage.data.Message;
+import com.udacity.jwdnd.course1.cloudstorage.services.FileService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+
+@Controller
+public class FileController {
+
+    @Autowired
+    private FileService fileService;
+
+
+    @PostMapping("/fileupload")
+    public String fileUpload(FileForm file, Model model) {
+
+       Message message = fileService.uploadFile(file);
+
+        model.addAttribute("message", message.getMsg());
+        model.addAttribute("success", message.isSuccess());
+        model.addAttribute("error", message.isError());
+
+        return "result";
+    }
+
+    @GetMapping("/file/delete")
+    public String fileDelete(@RequestParam("id") Integer id, Model model) {
+
+        Message message = fileService.deleteFile(id);
+
+        model.addAttribute("message", message.getMsg());
+        model.addAttribute("success", message.isSuccess());
+        model.addAttribute("error", message.isError());
+
+        return "result";
+    }
+
+    @GetMapping("/file/download")
+    public void fileDownload(@RequestParam("id") Integer id, HttpServletResponse response, Model model) {
```

Comment:
Make the return type to `String` so that it will return results page.
```suggestion
    public String fileDownload(@RequestParam("id") Integer id, HttpServletResponse response, Model model) {
```